pkg/occlient/occlient.go

```diff
@@ -63,26 +64,17 @@ import (
 	"k8s.io/client-go/util/retry"
 )
 
-// CreateType is an enum to indicate the type of source of component -- local source/binary or git for the generation of app/component names
-type CreateType string
-
-const (
-	// GIT as source of component
-	GIT CreateType = "git"
-	// LOCAL Local source path as source of component
-	LOCAL CreateType = "local"
-	// BINARY Local Binary as source of component
-	BINARY CreateType = "binary"
-	// NONE indicates there's no information about the type of source of the component
-	NONE CreateType = ""
+var (
+	DEPLOYMENT_CONFIG_NOT_FOUND_ERROR_STR string = "deploymentconfigs.apps.openshift.io \"%s\" not found"
```

Comment:
Just out of curiosity. Why this constant exists?

Reply:
It seems to be used for checking error returned https://github.com/redhat-developer/odo/pull/1360/files#diff-6025e9b620e9321c03c11bf29e5c4dbdR2714

Reply:
Yes, I saw it there. But why do we need a variable that is just a description of an error that is used in one place?

Reply:
just to avoid hard coding it to a function that only gets dc from its name I have it there so that the var can be used in future to support functions for example that get dc using a different/combination of selectors but in essence any similar but slightly different function...

Reply:
I'm still not sure how is this case different from any other error messages that we have elsewhere :-(

Reply:
in this case we are _catching an error using the error message and acting differently_ in all other cases we are just unwinding and returning the errors.
Maybe we can have a pattern similar to `os.IsNotExist(err)` which takes in the error and based on the message do different actions @anmolbabu . This will hide this variable.
Atleast we can make this a const? `Sprintf` generates a new string and hence wouldn't be a project.